solara/website/pages/documentation/advanced/content/10-howto/70-make-a-vue-component.md

```diff
@@ -0,0 +1,196 @@
+
+
+## Loading the confetti library.
+
+Now that we have our skeleton setup, we can add the confetti library. We will use the [`canvas-confetti`](https://www.npmjs.com/package/canvas-confetti) library, which is available on a CDN. We will add this to our `button-confetti.vue` file with unfortunately a bit of boilerplate code to load the library.:
+
+```vue
+<template>
+    <button class="button-confetti" @click="clickHandler">
+        {{label}}
+    </button>
+</template>
+<script>
+module.exports = {
+    created() {
+        console.log("button-confetti: created");
+        this.loadedConfetti = this.loadConfetti();
+    },
+    mounted() {
+        console.log("button-confetti: mounted");
+    },
+    watch: {
+        label() {
+            console.log("button-confetti: label changed");
+        }
+    },
+    methods: {
+        clickHandler() {
+            if(this.click) {
+                this.click({extra: 'foo'})
+            }
+            this.showConfetti();
+        },
+        async showConfetti() {
+            // make sure it is loaded by waiting on the Promise
+            await this.loadedConfetti;
+            await new Promise((resolve) => setTimeout(resolve, 1000))
+            confetti();
+        },
+        /* begin boilerplate */
+        async loadConfetti() {
+            let confettiUrl = "https://cdn.jsdelivr.net/npm/canvas-confetti@1.9.3/dist/confetti.browser.min.js";
+            require.config({
+                map: {
+                    '*': {
+                        'confetti': confettiUrl,
+                    }
+                }
+            })
+            await new Promise((resolve, reject) => {
+                require(['confetti'], (confettiModule) => {
+                    resolve(confettiModule);
+                }, reject)
+            });
+        },
+        /* end boilerplate */
+    },
+}
+</script>
+<style id="button-confetti">
+    .button-confetti {
+        background-color: lightblue;
+        margin: 10px;
+        padding: 4px;
+        border: 1px solid black;
+    }
+</style>
+```
+
+[<img src="https://py.cafe/logos/pycafe_logo.png" alt="PyCafe logo" width="24" height="24"> Run and edit this example at PyCafe](https://py.cafe/maartenbreddels/solara-howto-component-vue-B).
+
+
+## Triggering the confetti from the Python side
+
+Although we have a working button showing out confetti, this is triggered by a button in our vue template. There are situations where we want to trigger the confetti from the Python side. We can do this by making our
+Vue component respond to its argument in the `watch` section. We also remove any vue from out template so that our vue template becomes a non-visual component.
```

Comment:
```suggestion
Vue component respond to its argument in the `watch` section. We also remove any vue from our template so that our vue template becomes a non-visual component.
```